packages/trader/src/AppV2/Components/PurchaseButton/purchase-button-content.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import clsx from 'clsx';
+import { CaptionText } from '@deriv-com/quill-ui';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { getLocalizedBasis } from '@deriv/shared';
+import { Money } from '@deriv/components';
+
+type TPurchaseButtonContent = {
+    current_stake?: number | null;
+    info: ReturnType<typeof useTraderStore>['proposal_info'][0];
+    is_reverse?: boolean;
+} & Pick<
+    ReturnType<typeof useTraderStore>,
+    | 'currency'
+    | 'has_open_accu_contract'
+    | 'is_accumulator'
+    | 'is_multiplier'
+    | 'is_vanilla_fx'
+    | 'is_vanilla'
+    | 'is_turbos'
+>;
+
+const PurchaseButtonContent = ({
+    currency,
+    current_stake,
+    has_open_accu_contract,
+    info,
+    is_accumulator,
+    is_multiplier,
+    is_turbos,
+    is_vanilla,
+    is_vanilla_fx,
+    is_reverse,
+}: TPurchaseButtonContent) => {
+    const localized_basis = getLocalizedBasis();
```

Comment:
could use some destructuring here

Reply:
Hi! I'll be as a future improvement, which I do in my next card, if you don't mind. Don't want to add changes in current PR